tests/functional/test_install.py

```diff
@@ -2371,14 +2372,68 @@ def test_install_logs_pip_version_in_debug(
     assert_re_match(pattern, result.stdout)
 
 
-def test_install_dry_run(script: PipTestEnvironment, data: TestData) -> None:
-    """Test that pip install --dry-run logs what it would install."""
-    result = script.pip(
-        "install", "--dry-run", "--find-links", data.find_links, "simple"
-    )
+@pytest.fixture
+def install_find_links(
+    script: PipTestEnvironment,
+    data: TestData,
+) -> Callable[[Iterable[str], bool, Optional[Path]], TestPipResult]:
+    def install(
+        args: Iterable[str], dry_run: bool, target_dir: Optional[Path]
+    ) -> TestPipResult:
+        return script.pip(
+            "install",
+            *(
+                (
+                    "--target",
+                    str(target_dir),
+                )
+                if target_dir is not None
+                else ()
+            ),
+            *(("--dry-run",) if dry_run else ()),
+            "--no-index",
+            "--find-links",
+            data.find_links,
+            *args,
+        )
+
+    return install
```

Comment:
Making `install_find_links` into a fixture rather than a simple helper function (that takes script and data as additional arguments) is very confusing to me. I'm a bug fan of keeping things simple. Do we need this extra indirection?

Reply:
Totally agree. I have fixed this!